src/components/viewerJs/viewerJs.js

```diff
@@ -0,0 +1,91 @@
+const STATE_TOPIC_URL = 'pdfUrl';
+
+const loadURL = function(url){
+	FSBL.Clients.Logger.log("Loading URL: " + url);
+	if (!window.location.href.includes(url)){
+		FSBL.Clients.WindowClient.setComponentState({ field: STATE_TOPIC_URL, value: url });
+		window.location = window.location.protocol + "//" 
+						+ window.location.hostname 
+						+ (window.location.port ? ":"+window.location.port : "") 
+						+ window.location.pathname + "#" 
+						+ url;
+		window.location.reload();
+	}
+	FSBL.Clients.WindowClient.bringWindowToFront();
+}
+
+const incomingContextHandler = function(linkerData){
+	//Support linker messages on the url and pdf channels
+	if (linkerData && (typeof linkerData == "string" || linkerData.url || linkerData.pdf)) {
+		let linkerUrl = typeof linkerData == "string" ? linkerData : 
+						linkerData.url ? linkerData.url : linkerData.pdf;
+		FSBL.Clients.Logger.log("Got linker data URL: " + linkerUrl);
+		loadURL(linkerUrl);
+	} else {
+		FSBL.Clients.Logger.error("No PDF URL to load from linker");
+	}
+};
+
+const setupContextSharing = function (){
+	//register linker topics
+	FSBL.Clients.LinkerClient.subscribe("url", incomingContextHandler);
+	FSBL.Clients.LinkerClient.subscribe("pdf", incomingContextHandler);
+
+	//support Drag and Drop
+	FSBL.Clients.DragAndDropClient.addReceivers({
+		receivers: [
+			{
+				type: 'pdf',
+				handler: function (err, response) {
+					if (!err) { incomingContextHandler(response.data); }
+					else { FSBL.Clients.Logger.error("Error on drag and drop (pdf): ", err); }
+				}
+			},
+			{
+				type: 'url',
+				handler: function (err, response) {
+					if (!err) { incomingContextHandler(response.data); }
+					else { FSBL.Clients.Logger.error("Error on drag and drop (url): ", err); }
+				}
+			}
+		]
+	});
+};
+
+const init = function (){
+	//Check for existing component state
+	let state = 
```

Comment:
Seems that **state** is set but then doesn't seem to be read anywhere, is it needed?

Reply:
Yep can be removed